server/middleware/appender.go

```diff
@@ -0,0 +1,109 @@
+package middleware
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/src-d/code-annotation/server/serializer"
+
+	"github.com/src-d/code-annotation/server/service"
+)
+
+func NewMiddlewaresWithAppender(ms []func(http.Handler) http.Handler) []func(http.Handler) http.Handler {
+	return append(ms, Appender)
+}
+
+func Appender(next http.Handler) http.Handler {
```

Comment:
we definitely need comments here. It isn't obvious what `Appender` does

Reply:
Yes, you're right.
And the name is not clear at all.
Options:
- now,
- merge and document.

Anyhow, I planned to add it to the login/callback so all errors, logging... will be homogeneous

Reply:
I think we need to document it now. Because I don't understand it completely even for review.
Like what would happen if I use `w` as a normal writer. Looks like it will just fail even if you implement `http.ResponseWriter` methods instead of current panic.
Is it normal behavior? Is it a bug?
Without documentation I don't know.

Reply:
If you're using that middleware, the w you receive won't accept writes.
I'm writing a couple of lines with doc; I'll ping you in a few mins when ready :D

Reply:
then it's completely incorrect code.
handler accepts `http.ResponseWriter`. But it isn't `http.ResponseWriter` anymore.

Reply:
functionality dropped